### PR TITLE
Cleanups of the storage definitions.

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

In PR #2896 it was suggested to move the definition of the stores to `mod.rs`. Ultimately, this is not what is done in this PR as there was disagreement.

## Proposal

The following was done:
* Simplify out the relevant code from the stores, in particular the `LocalStackTestContext` that is no longer used.
* Remove unused enum in the error types, that is `MissingDatabase` and `AlreadyExistingDatabase`.
* Introduce constructor for the `RocksDbStoreConfig` and similar.
* Remove the `pub` attribute when not needed.
* Correct the mis-attribution "lru splitting" into "lru caching".
* Add a `PathWithStorage` function and remove another function.
* Put the constructor definition of the `VISIBLE_MAX_VALUE_SIZE` for DynbamoDb and remove the test that simply reproduce the formula.
* The `InvalidTableName` was renamed as `InvalidNamespace`.

## Test Plan

CI as the functionality should stay exactly the same.

## Release Plan

Not relevant.

## Links

None.

### CODE REVIEW

File: linera-views/src/backends/mod.rs [hunk 1]
```diff
@@ -15,13 +15,180 @@ pub mod lru_caching;
...
+            cache_size: common_config.cache_size,
+        }
+    }
+}
+
+/// The combined error type for the `ScyllaDbStore`.
+#[cfg(with_scylladb)]
+pub type ScyllaDbStoreError = crate::value_splitting::ValueSplittingError<
+    crate::backends::scylla_db::ScyllaDbStoreInternalError,
+>;
```
Comment (afck) [inline linera-views/src/backends/mod.rs:194]: We could still have these in separate submodules and re-export them, to reduce the number of `cfg` statements.

In general I'm not sure if moving these directly into `backends` is an improvement.

**Review by afck — COMMENTED** [review]
I like the simplifications using the constructors (like `DynamoDbStoreConfig::new`). :+1: 

But I'd prefer to keep types specific to different backends in different modules, personally.

**Comment by MathieuDutSik** [comment]
> I like the simplifications using the constructors (like `DynamoDbStoreConfig::new`). 👍
> 
> But I'd prefer to keep types specific to different backends in different modules, personally.

Ok, we will have to see what @ma2bd thinks as it was his initial suggestion. I like the separation.

**Comment by MathieuDutSik** [comment]
> I like the simplifications using the constructors (like `DynamoDbStoreConfig::new`). 👍
> 
> But I'd prefer to keep types specific to different backends in different modules, personally.

That is indeed a little problematic. So, I revamped the PR by focusing on less controversial issues that hopefully could be accepted more easily.

**Review by afck — APPROVED** [review]

**Pull request merged**